apisix/plugins/redirect.lua

```diff
@@ -148,7 +150,30 @@ function _M.rewrite(conf, ctx)
     core.log.info("plugin rewrite phase, conf: ", core.json.delay_encode(conf))
 
     local ret_code = conf.ret_code
-    local ret_port = tonumber(ctx.var["var_x_forwarded_port"])
+    local ret_port = nil
+    local attr = plugin.plugin_attr(plugin_name)
```

Comment:
Where did you define the variable `plugin_name`?

Reply:
here:
https://github.com/apache/apisix/blob/6ebe02797fb564c84b30df1865adef7f473880bf/apisix/plugins/redirect.lua#L65-L67

Reply:
Was defined before :)